src/components/Vault/hooks/useVaultsTotalPrices.ts

```diff
@@ -0,0 +1,18 @@
+import { useVaults } from "hooks/useVaults";
+import { IVault } from "types/types";
+import { fromWei } from "utils";
+
+export function useVaultsTotalPrices(vaults: IVault[]) {
+  const { tokenPrices } = useVaults();
+
+  const totalPrices: { [token: string]: number } = {};
+
+  vaults.forEach(vault => {
```

Comment:
this case really fits reduce method

Reply:
Mm not sure I understand. In that case we want to traverse the vaults and only if we are able to calculate `totalUSDValue` then we add it to our object. We have no accumulated value here.

Reply:
please refer to reduce example from Honeypots.tsx